### PR TITLE
Added support for dns keys and dns policies

### PR DESCRIPTION
**Pull request by ajinkya1986**
1. Added DNS Keys and DNS Policies
2. Added indexes
3. Added test cases
4. Added cleanup jobs
5. All tests are passing

### CODE REVIEW

File: cartography/intel/gcp/dns.py [hunk 1]
```diff
@@ -121,6 +121,103 @@ def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[R
...
+    :rtype: list
+    :return: List of Resource Record Sets
+    """
+    try:
+        dns_keys = []
+        for zone in dns_zones:
+            request = dns.dnsKeys().list(project=project_id, managedZone=zone['name'])
+            while request is not None:
+                response = request.execute()
+                if response.get('dnsKeys'):
```
Comment (mpurusottamc) [inline cartography/intel/gcp/dns.py:147]: we should avoid nested loops like this. Instead, return all the keys from this method and create a separate trahsform method to assign zone & id values.

File: cartography/intel/gcp/dns.py [hunk 2]
```diff
@@ -121,6 +121,103 @@ def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[R
...
+    :rtype: list
+    :return: List of DNS Policies
+    """
+    try:
+        policies = []
+        request = dns.policies().list(project=project_id)
+        while request is not None:
+            response = request.execute()
+            if response.get('policies'):
+                for policy in response['policies']:
```
Comment (mpurusottamc) [inline cartography/intel/gcp/dns.py:187]: same as above. let's create a transform method and move the logic to populate additional fields into that method.

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
transform related logic should be moved out of get or list objects method.

File: cartography/intel/gcp/dns.py (hunk 1 above)
Comment (ajinkya1986) [reply cartography/intel/gcp/dns.py:147]: There would be transform method for each resource and then some of these methods will also have loops and some will have to populate only one field.

File: cartography/intel/gcp/dns.py (hunk 2 above)
Comment (ajinkya1986) [reply cartography/intel/gcp/dns.py:187]: Like for this policy we are populating only the `id`. The transform method will be very small

File: cartography/intel/gcp/dns.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def get_dns_zones(dns: Resource, project_id: str, common_job_parameters) -> List
 
 
 @timeit
-def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[Resource]:
+def get_dns_rrs(dns: Resource, zone: list, project_id: str) -> List[Resource]:
```
Comment (mpurusottamc) [inline cartography/intel/gcp/dns.py:79]: `list` is invalid, it should be `List`. Also, in this case, i think, it's a Dictionary and not a List.

File: cartography/intel/gcp/dns.py [hunk 4]
```diff
@@ -121,6 +114,134 @@ def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[R
             # raise
             return []
 
+@timeit
+def transform_rrs(rrsets: list, zone: dict, project_id: str):
```
Comment (mpurusottamc) [inline cartography/intel/gcp/dns.py:118]: `list` or `dict` are invalid. Are the tests working properly? I will guess that Tests are failing. Please ensure all the tests are working properly.

File: cartography/intel/gcp/dns.py [hunk 5]
```diff
@@ -121,6 +114,134 @@ def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[R
...
+            resource_name='dns_resource_record_set', project_id=project_id, dns_zone_name=zone['name'], dns_rrset_name=resource_record_set['name'])
+        resource_record_set[
+            "id"
+        ] = f"projects/{project_id}/resourceRecordSet/{resource_record_set.get('name',None)}"
+        list_rrs.append(resource_record_set)
+
+    return list_rrs
+
+@timeit
+def get_dns_keys(dns: Resource, zone: dict, project_id: str) -> List[Resource]:
```
Comment (mpurusottamc) [inline cartography/intel/gcp/dns.py:133]: Please check it throughout the code. I see `dict` here again. It should be `Dict`.

File: cartography/intel/gcp/dns.py [hunk 6]
```diff
@@ -121,6 +114,134 @@ def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[R
...
+                    "Could not retrieve DNS Keys on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return []
+        else:
+            # raise
+            return []
+
+@timeit
+def transform_dns_keys(dnsKeys: list, project_id: str, zone: dict):
```
Comment (mpurusottamc) [inline cartography/intel/gcp/dns.py:172]: Same as above.

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
There are few changes needed. Please re-run unit tests to ensure code is working properly.

**Review by mpurusottamc — APPROVED** [review]
looks good.

**Pull request merged**